generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -8,80 +8,116 @@
 import com.scottlogic.deg.generator.utils.FileUtils;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.ArrayList;
 
 /**
  * Class used to determine whether the command line options are valid for generation
  */
-public class GenerationConfigValidator {
+public class GenerationConfigValidator implements ConfigValidator {
 
     private final FileUtils fileUtils;
-    private final GenerationConfigSource configSource;
-    private final OutputTarget outputTarget;
 
     @Inject
-    public GenerationConfigValidator(GenerationConfigSource configSource,
-                                     OutputTarget outputTarget,
-                                     FileUtils fileUtils) {
-        this.configSource = configSource;
-        this.outputTarget = outputTarget;
+    public GenerationConfigValidator(FileUtils fileUtils) {
         this.fileUtils = fileUtils;
     }
 
-    public ValidationResult validateCommandLinePreProfile(GenerationConfig config) {
+    @Override
+    public ValidationResult preProfileChecks(GenerationConfig config, GenerationConfigSource generationConfigSource) {
         ArrayList<String> errorMessages = new ArrayList<>();
-        ValidationResult validationResult = new ValidationResult(errorMessages);
 
-        if (config.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM
-            && !config.getMaxRows().isPresent()) {
+        checkSwitches(config, errorMessages);
 
-            errorMessages.add("RANDOM mode requires max row limit: use -n=<row limit> option");
-        }
+        checkProfileInputFile(errorMessages, generationConfigSource.getProfileFile());
 
-        return validationResult;
+        return new ValidationResult(errorMessages);
     }
 
-    public ValidationResult validateCommandLinePostProfile(Profile profile) {
+    @Override
+    public ValidationResult postProfileChecks(Profile profile,
+                                              GenerationConfigSource configSource,
+                                              OutputTarget outputTarget) {
         ArrayList<String> errorMessages = new ArrayList<>();
         ValidationResult validationResult = new ValidationResult(errorMessages);
 
-        if (configSource.shouldViolate()) {
-            checkViolationGenerateOutputTarget(errorMessages, outputTarget, profile.rules.size());
-        } else {
-            checkGenerateOutputTarget(errorMessages, outputTarget);
+        if (outputTarget instanceof FileOutputTarget ) {
+            if (configSource.shouldViolate()) {
+                checkViolationGenerateOutputTarget(
+                    errorMessages,
+                    configSource,
+                    (FileOutputTarget)outputTarget,
+                    profile.rules.size()
+                );
+            } else {
+                checkGenerateOutputTarget(errorMessages, configSource, (FileOutputTarget)outputTarget);
+            }
         }
 
         return validationResult;
```

Comment:
This could be changed to match the above :

```suggestion
         return new ValidationResult(errorMessages);
```